Fixes authentication checks and refactors logout

dhsaks opened the pull request:
- In server code, under account and admin endpoints, config was defined twice causing server to silently ignore authentication rules. Now, admin and account pages should be blocked to unauthenticated users.
- Refactored types for passing logout data within react and redux

Commit:
Fixes authentication checks and refactors logout

- In server code, under account and admin endpoints, config was defined twice causing server to silently ignore authentication rules

- Refactored types for passing logout data within react and redux

ozguruksal commented on src/global/client/pages/login/components/Logout/Logout.tsx:
```diff
@@ -13,6 +13,9 @@ import './_Logout.scss';
+    doLogout
 } from '../../actions';
 
 import jsonFetch, { IJSONFetch } from '../../../../api/jsonfetch'
```
I think, we should have simpler paths like "jsonfetch" instead of relative paths or full paths like "../../../..jsonfetch" or "src/blah/blah/jsonfetch"; 
Otherwise, it will be painful. I will make a proposal on issues section if I find a solution.

Commit:
Removed dead code

Per https://github.com/hydrotik/node-hapi-react-redux-sass-typescript-mongo-webpack-hmr-gulp/pull/115

ghost commented:
Yeah, that line is unnecessary, so I've removed it
